core/io/dir_access.cpp

```diff
@@ -35,13 +35,16 @@
 #include "core/os/os.h"
 #include "core/os/time.h"
 #include "core/templates/local_vector.h"
+#include "editor/settings/editor_settings.h"
```

Comment:
This breaks encapsulation and should be avoided if possible, core code shouldn't include editor code

But if it is to be included it has to be guarded with `TOOLS_ENABLED`